File: Java/やさしいJava/11/sample4.java
class car{
	protected int num;
	protected double gass;
	
	public car(){
		System.out.println("車を作成しました。");
		num = 0;
		gass =0.0;
		System.out.println("ナンバーを"+num+"に"+"ガソリン量を"+gass+"にしました。");
	}
	
	public car(int n , double m){
		System.out.println("車を作成しました。");
		num = n;
		gass =m;
		System.out.println("ナンバーを"+num+"に"+"ガソリン量を"+gass+"にしました。");
	}	
	
	public String toString(){
		String str = "ナンバー"+num+"ガソリン量"+gass+"の車です。";
		return str;
	}
}
	
class sample4{
	public static void main(String[] args){
		car car1 = new car(1234,20.5);
		car car2 = new car();
		
		System.out.println(car1);
		System.out.println(car2);
		
	}
}

File: C言語/やさしいC/Lesson11 構造体,共用体,列挙型/Sample4.c
#include <stdio.h>
/*
構造体のポインタを関数に引き渡すときはアロー関数を使う。
ちなみにポインタではなく構造体をそのまま渡すとやっぱり直渡しになる。
メンバをそれぞれコピーして引き渡される。
*/
typedef struct Car {
    int num;
    double gas;
} Car;
void show(Car *pC);

int main() {
    Car car1 = {3, 3.4};
    show(&car1);
    return 0;
}
void show(Car *pC) {
    //アロー関数
    printf("num: %d, gas: %lf", pC -> num,pC -> gas);
}
